File: packages/android/src/main/java/com/kozmos/components/MapOverlay/MapOverlay.kt
package com.kozmos.components.MapOverlay

import com.kozmos.tokens.KozmosColors
import com.kozmos.tokens.KozmosDimensions

import androidx.compose.foundation.layout.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp

enum class OverlayPosition {
    TOP_LEFT, TOP_RIGHT, BOTTOM_LEFT, BOTTOM_RIGHT, TOP_CENTER, BOTTOM_CENTER
}

@Composable
fun KozmosMapOverlay(
    modifier: Modifier = Modifier,
    position: OverlayPosition = OverlayPosition.TOP_LEFT,
    content: @Composable () -> Unit
) {
    Box(
        modifier = modifier
            .fillMaxSize()
            // Translates `pointer-events-none` into native transparent interaction bounds
            // Ensures Android System Navigation Bars do not collide with absolute components
            .windowInsetsPadding(WindowInsets.safeDrawing)
    ) {
        val alignment = when (position) {
            OverlayPosition.TOP_LEFT -> Alignment.TopStart
            OverlayPosition.TOP_RIGHT -> Alignment.TopEnd
            OverlayPosition.BOTTOM_LEFT -> Alignment.BottomStart
            OverlayPosition.BOTTOM_RIGHT -> Alignment.BottomEnd
            OverlayPosition.TOP_CENTER -> Alignment.TopCenter
            OverlayPosition.BOTTOM_CENTER -> Alignment.BottomCenter
        }

        // Structural map geometry clearance (MapLibre telemetry / Compass bounds)
        val paddingModifier = when (position) {
            OverlayPosition.BOTTOM_LEFT, OverlayPosition.BOTTOM_RIGHT, OverlayPosition.BOTTOM_CENTER -> 
                Modifier.padding(bottom = KozmosDimensions.primitivesLayoutSpacing600, start = KozmosDimensions.primitivesLayoutSpacing200, end = KozmosDimensions.primitivesLayoutSpacing200, top = KozmosDimensions.primitivesLayoutSpacing200)
            else -> Modifier.padding(KozmosDimensions.primitivesLayoutSpacing200)
        }

        Box(
            modifier = Modifier
                .align(alignment)
                .then(paddingModifier)
                .widthIn(max = 384.dp) // `md:w-96` standard bounds translation natively
        ) {
            content()
        }
    }
}

File: packages/ios/Sources/Components/MapOverlay/MapOverlay.swift
import SwiftUI

public enum KozmosOverlayPosition {
    case topLeft, topRight, bottomLeft, bottomRight, topCenter, bottomCenter
}

public struct KozmosMapOverlay<Content: View>: View {
    public var position: KozmosOverlayPosition
    public var content: () -> Content

    public init(position: KozmosOverlayPosition = .topLeft, @ViewBuilder content: @escaping () -> Content) {
        self.position = position
        self.content = content
    }

    public var body: some View {
        let alignment: Alignment = {
            switch position {
            case .topLeft: return .topLeading
            case .topRight: return .topTrailing
            case .bottomLeft: return .bottomLeading
            case .bottomRight: return .bottomTrailing
            case .topCenter: return .top
            case .bottomCenter: return .bottom
            }
        }()

        // ZStack mathematically mimics `absolute z-50` bounds natively isolating domain components
        ZStack(alignment: alignment) {
            KozmosColors.primitivesColorsBackground0.opacity(0.0001) // Transparent hit box forcing strict ZStack global projection coordinates 
                .ignoresSafeArea()
                .allowsHitTesting(false) // CRITICAL: Ensures the projection layer DOES NOT intercept map gestures!
            
            content()
                .padding(.horizontal, KozmosDimensions.primitivesLayoutSpacing200)
                .padding(.top, position == .topLeft || position == .topRight || position == .topCenter ? KozmosDimensions.primitivesLayoutSpacing200 : KozmosDimensions.primitivesLayoutSpacing0)
                // Adds extra structural padding to evade iOS Home Indicator AND MapKit / MapLibre compass icons
                .padding(.bottom, position == .bottomLeft || position == .bottomRight || position == .bottomCenter ? KozmosDimensions.primitivesLayoutSpacing600 : KozmosDimensions.primitivesLayoutSpacing200)
                .frame(maxWidth: 384) // `md:w-96` analog limits
        }
    }
}
